languages/cpp/tree-sitter/Parse_cpp_tree_sitter.ml

```diff
@@ -619,10 +619,27 @@ let map_preproc_def (env : env) ((v1, v2, v3, v4) : CST.preproc_def) =
   let v2 = str env v2 (* pattern [a-zA-Z_]\w* *) in
   let v3 =
     match v3 with
-    | Some tok ->
-        let x = token env tok (* preproc_arg *) in
-        (* TODO: we should parse this x! can be DefineExpr, etc. *)
-        DefineTodo ("MacroBody", x)
+    | Some tok -> (
+        let parse_number_literal t s =
+          match Common2.int_of_string_c_octal_opt s with
+          | Some i -> Some (C (Int (Some i, t)))
+          | None -> (
+              match float_of_string_opt s with
+              | Some f -> Some (C (Float (Some f, t)))
+              | None -> None)
+        in
+        let remove_comment s =
+          match Str.split (Str.regexp "//\\|/\\*") s with
+          | s :: _ -> Some (s |> String.trim)
+          | [] -> None
+        in
+        (* TODO: parse exprs other than number literals. *)
+        match token env tok (* preproc_arg *) with
```

Comment:
we might want to instead change the tree-sitter-cpp grammar and use GLR to try first
to parse the preproc_arg as an expression, and only if it fails use preproc_arg,
so we would not need to do some parsing work here. We had a similar issue with Rust macros where @brandonspark also did some extra reparsing in Parse_rust_tree_sitter.ml which might be better done by tree-sitter itself.